src/main/java/core/org/akaza/openclinica/service/DiscrepancyNoteUtil.java

```diff
@@ -128,6 +128,7 @@ public void injectParentDiscNotesIntoDisplayStudyEvents(List<DisplayStudyEventBe
         for (DisplayStudyEventBean dStudyEventBean : displayStudyBeans) {
             studyEventBean = dStudyEventBean.getStudyEvent();
             // All EventCRFs for a study event
+            eventCrfDAO = new EventCRFDAO(dataSource);
```

Comment:
we shouldn't be initializing DAOs like this anymore. Instead do something like below:
eventCRFDAO = (EventCRFDAO) SpringServletAccess.getApplicationContext(context).getBean("eventCRFJDBCDao");

See Jon's email for further information about this.

Reply:
Even with the Qualifier annotation the eventCRFDAO still null, made some modification on local but it still null so I use the easiest way that you suggest bypassing the ServletContext to the method and use it to get the DAO.